### PR TITLE
Actions: Add github actions

### PR DESCRIPTION
**Pull request by peterwillcn**
Signed-off-by: Xiaobo Tian <peterwillcn@gmail.com>

Maintainer: peterwillcn@gmail.com
Compile tested: x86, x86_64, OpenWrt master
Run tested: x86_64

Description:
Add github actions check pr

### CODE REVIEW

**Comment by jneuhauser** [comment]
An why should this action make sense for the packages repo?
You only build the firmware which contains only the same packages as the snapshot builds and there is no package from this packages repo included.

And there is already a working code check for pr's implemented with circleci.

**Comment by peterwillcn** [comment]
> An why should this action make sense for the packages repo?
> You only build the firmware which contains only the same packages as the snapshot builds and there is no package from this packages repo included.
> 
> And there is already a working code check for pr's implemented with circleci which builds only the changed packages with a prebuilt sdk.

I think make sure that all the packages will compile under openwrt

**Comment by neheb** [comment]
The more the merrier honestly. I've been wanting to see multiple platforms being tested with CircleCI. Not sure if this can do it.

File: .github/workflows/build_x86_64.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          google* \
+          dotnet* \
+          powershell \
+          openjdk* \
+          mysql* \
+          php*
+        sudo -E apt-get update
+        sudo -E apt-get -y install build-essential asciidoc binutils bzip2 gawk gettext git libncurses5-dev patch unzip zlib1g-dev lib32gcc1 libc6-dev-i386 subversion flex node-uglify gcc-multilib p7zip p7zip-full msmtp libssl-dev texinfo libglib2.0-dev xmlto qemu-utils upx-ucl libelf-dev autoconf automake libtool autopoint device-tree-compiler libuv-dev
+        sudo -E apt-get -y autoremove --purge
+        sudo -E apt-get clean
```
Comment (champtar) [inline .github/workflows/build_x86_64.yml:51]: Is there a way to use a container ? so we avoid reinstalling the base package each time.

File: .github/workflows/build_x86_64.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  #schedule:
+  #  - cron: 0 8 * * 5
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@master
```
Comment (champtar) [inline .github/workflows/build_x86_64.yml:23]: Do I understand correctly that you test the master branch on pull request ?

Comment (champtar) [reply .github/workflows/build_x86_64.yml:23]: Also you don't seem to use this checkout at all

File: .github/workflows/build_x86_64.yml (hunk 1 above)
Comment (neheb) [reply .github/workflows/build_x86_64.yml:51]: Huh. Why purge all those packages?

Comment (peterwillcn) [reply .github/workflows/build_x86_64.yml]: > Is there a way to use a container ? so we avoid reinstalling the base package each time.

I want to change to buildbot's docker images, but I'm having trouble with it at the moment

Comment (peterwillcn) [reply .github/workflows/build_x86_64.yml]: > Huh. Why purge all those packages?

Saves space for compiling the image.

File: .github/workflows/build_x86_64.yml (hunk 2 above)
Comment (peterwillcn) [reply .github/workflows/build_x86_64.yml]: You're right, it's not really used here.

**Comment by BKPepe** [comment]
This https://github.com/P3TERX/Actions-OpenWrt looks very similar to your changes.

**Comment by BKPepe** [comment]
I see that you posted it in OpenWrt main repository ( https://github.com/openwrt/openwrt/pull/2867 ) , but the changes there looked to very similar to @P3TERX [repository](https://github.com/P3TERX/Actions-OpenWrt/). I am wondering if these changes in this PR are yours.

**Comment by peterwillcn** [comment]
> I see that you posted it in OpenWrt main repository ( [openwrt/openwrt#2867](https://github.com/openwrt/openwrt/pull/2867) ) , but the changes there looked to very similar to @P3TERX [repository](https://github.com/P3TERX/Actions-OpenWrt/). I am wondering if these changes in this PR are yours.

Yes, This is fork and MIT has modified the version to make it compile properly.

**Comment by Neustradamus** [comment]
Any news on Actions?

**Pull request closed without merging**